tests/test_self.py

```diff
@@ -1305,14 +1248,11 @@ def test_recursive_current_dir(self):
                 for path in sys.path
                 if not os.path.basename(path) == "regrtest_data"
             ]
-            with _test_cwd():
+            with tmpdir.as_cwd():
                 os.chdir(join(HERE, "regrtest_data", "directory", "subdirectory"))
```

Comment:
Let me know if I'm misunderstanding, but to me this looks like we're now creating a temporary directory, then cd'ing into it, then disregarding it and cd'ing into something else.

Reply:
Yes we're simply using the fact that the context manager will restore the current working directory. What do you think of @DanielNoord proposal to use out context manager instead ?

Reply:
Nice catch @jacobtylerwalls. I think this calls for another decorators `_restore_cwd()`:

```python
@contextlib.contextmanager
 def _restore_cwd( ) -> Generator[None, None, None]:
     original_dir = os.getcwd()
     try:
         yield
     finally:
         os.chdir(original_dir)
```

Reply:
That's exactly what we have, let me change this MR to use our current context manager instead of tmpdir then. (We may have to create the directory if it does not exists but let's see).

I opened #7175 and #7176 so this one is only about current working directory management.